test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetVerifyTests.cs

```diff
@@ -0,0 +1,150 @@
+                var result = _msbuildFixture.RunDotnet(
+                    packageDir,
+                    $"nuget verify {packageFile.FullName}",
+                    ignoreExitCode: true);
+
+                result.Success.Should().BeFalse(because: result.AllOutput);
+                result.Output.Should().Contain(_notSignedErrorCode);
+            }
+        }
+
+        [CIOnlyTheory]
+        [InlineData("--all")]
+        [InlineData("")]
+        public void Verify_AuthorSignedAndTimestampedPackage_Succceeds(string optionAll)
+        {
+            // Arrange
+            using (var testDirectory = TestDirectory.Create())
+            {
+                var packageFile = new FileInfo(Path.Combine(testDirectory, "TestPackage.AuthorSigned.1.0.0.nupkg"));
+                var package = SigningTestUtility.GetResourceBytes(packageFile.Name);
+                File.WriteAllBytes(packageFile.FullName, package);
+
+                //Act
+                var result = _msbuildFixture.RunDotnet(
+                    testDirectory,
+                    $"nuget verify {packageFile.FullName} {optionAll}",
+                    ignoreExitCode: true);
+
+                result.Success.Should().BeTrue(because: result.AllOutput);
+                result.Output.Should().NotContain(_noTimestamperWarningCode);
+                result.Output.Should().NotContain(_primarySignatureInvalidErrorCode);
+            }
+        }
+
+        [CIOnlyTheory]
+        [InlineData("--certificate-fingerprint 775AAB607AA76028A7CC7A873A9513FF0C3B40DF09B7B83D21689A3675B34D9A")]
+        [InlineData("--certificate-fingerprint ABC --certificate-fingerprint DEF")]
+        public void Verify_SignedPackageWithoutAllowedCertificate_Fails(string fingerprints)
+        {
+            // Arrange
+            using (var testDirectory = TestDirectory.Create())
+            {
+                var packageFile = new FileInfo(Path.Combine(testDirectory, "TestPackage.AuthorSigned.1.0.0.nupkg"));
+                var package = SigningTestUtility.GetResourceBytes(packageFile.Name);
+                File.WriteAllBytes(packageFile.FullName, package);
+
+                //Act
+                var result = _msbuildFixture.RunDotnet(
+                    testDirectory,
+                    $"nuget verify {packageFile.FullName} {fingerprints}",
+                    ignoreExitCode: true);
+
+                result.Success.Should().BeFalse(because: result.AllOutput);
+                result.AllOutput.Should().Contain(_noMatchingCertErrorCode);
+            }
+        }
+
+        [CIOnlyTheory]
+        [InlineData("--certificate-fingerprint 3F9001EA83C560D712C24CF213C3D312CB3BFF51EE89435D3430BD06B5D0EECE")]
+        [InlineData("--certificate-fingerprint ABC --certificate-fingerprint DEF " +
+            "--certificate-fingerprint 3F9001EA83C560D712C24CF213C3D312CB3BFF51EE89435D3430BD06B5D0EECE")]
+        public void Verify_SignedPackageWithAllowedCertificate_Succceeds(string fingerprints)
```

Comment:
The previous test that passes a fingerprint already tests that NuGet.CommandLine.Xplat passes the fingerprint argument in the `VerifyArgs` object to the `VerifyCommandRunner` instance. Therefore, this test that tests asserts that a different scenario with the same input partemers has a different result, it's a duplicate of a test I would expect in NuGet.Commands.Tests or NuGet.Commands.FuncTests.  Therefore, I believe the `_Fails` and `_Succeeds` tests are duplicating existing test scenarios and therefore are making our CI a little bit slower without increasing test coverage.

Reply:
I just found that `VerifyCommandRunner` has tests only in [`NuGet.CommandLine.FuncTest`](https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/VerifyCommandTests.cs) but not in either `NuGet.Commands.Tests` or `NuGet.Commands.FuncTests`. This comment made me think a bit more about the purpose of `Dotnet.Integration.Tests`. As you mentioned in another conversation, dotnet integration tests are meant to ensure that different types work correctly when integrated into `dotnet`.

Changed the test to act only on multiple fingerprints. Will push in my next commit.

Reply:
I have created an [issue](NuGet/Home#10047) to add `unit` and `Func` tests for `VerifyCommandRunner `. We can remove few `dotnet integration` tests while addressing new issue.